packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -728,13 +731,57 @@ export async function finalizeTransformationJob(status: string) {
 export async function getValidLanguageUpgradeCandidateProjects() {
     const openProjects = await getOpenProjects()
     const javaMavenProjects = await validateOpenProjects(openProjects)
+    getLogger().info(`CodeTransformation: found ${javaMavenProjects.length} projects eligible for language upgrade`)
     return javaMavenProjects
 }
 
 export async function getValidSQLConversionCandidateProjects() {
-    const openProjects = await getOpenProjects()
-    const javaProjects = await getJavaProjects(openProjects)
-    return javaProjects
+    const embeddedSQLProjects: TransformationCandidateProject[] = []
+    await telemetry.codeTransform_validateProject.run(async () => {
+        telemetry.record({
+            codeTransformSessionId: CodeTransformTelemetryState.instance.getSessionId(),
+        })
+        const openProjects = await getOpenProjects()
+        const javaProjects = await getJavaProjects(openProjects)
+        let errorLog = ''
+        const isWindows = isWin()
+        const command = isWindows ? 'findstr' : 'grep'
```

Comment:
vscode ships with `rg` (ripgrep, extremely fast and cross-platform). Can we use that? https://github.com/microsoft/vscode-ripgrep

I don't exactly how or if it's exposed. I couldn't find an official vscode api. It looks like `findTextInFiles` is a proposed API: https://github.com/microsoft/vscode/issues?q=is%3Aissue%20state%3Aopen%20findTextInFiles

While we're waiting on that, at minimum, please extract this file-search logic to our shared `workspaceUtils.ts` module. This kind of functionality isn't something that should be casually added to an arbitrary Q module.

By extracting to a shared module, that gives us an interface that allows us to later switch it to `rg` (which we could sniff out in vscode's path somewhere, since it ships with `rg`), or `findTextInFiles` when that lands.